editor/lang/token.hh

```diff
@@ -33,20 +31,78 @@ public:
     // Other
     NUMBER,
     IDENTIFIER,
+    NONE
   };
 
   Token(Type type, int line, std::string lexeme, std::string literal)
       : m_type(type), m_line(line), m_lexeme(lexeme), m_literal(literal) {}
 
+  Token() : Token(NONE, 0, "", "") {}
+
+  Token(Type type, std::string lexeme) : m_type(type), m_lexeme(lexeme) {}
+
   friend std::ostream &operator<<(std::ostream &stream, const Token &token) {
-    stream << token.m_lexeme << "," << magic_enum::enum_name(token.m_type);
+	stream << token.m_lexeme << "," << token.m_type;
 
     return stream;
   }
 
+  void set_token_type(Type type) {
+    m_type = type;
+
+    // Find better way !!
+    if (type == Token::Type::IF) {
+      m_literal = "if";
```

Comment:
Why are you editing the `m_literal` instead of the `m_type`?

Also, I think they should be ALL CAPS.